internal/provider/resource_bootstrap_git.go

```diff
@@ -0,0 +1,1183 @@
+					planmodifiers.DefaultBoolValue(defaultOpts.NetworkPolicy),
+				},
+			},
+			"registry": {
+				Description: "Container registry where the toolkit images are published.",
+				Type:        customtypes.URLType,
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					planmodifiers.DefaultStringValue(defaultOpts.Registry),
+				},
+			},
+			"toleration_keys": {
+				Description: "List of toleration keys used to schedule the components pods onto nodes with matching taints.",
+				Type: types.SetType{
+					ElemType: types.StringType,
+				},
+				Optional: true,
+				Validators: []tfsdk.AttributeValidator{
+					setvalidator.ValuesAre(
+						stringvalidator.RegexMatches(regexp.MustCompile(rfc1123LabelRegex), rfc1123LabelError),
+						stringvalidator.LengthAtMost(63),
+					),
+				},
+			},
+			"watch_all_namespaces": {
+				Description: "If true watch for custom resources in all namespaces.",
+				Type:        types.BoolType,
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					planmodifiers.DefaultBoolValue(defaultOpts.WatchAllNamespaces),
+				},
+			},
+			"url": {
+				Description: "Url of git repository to bootstrap from.",
+				Type:        customtypes.URLType,
+				Required:    true,
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					resource.RequiresReplace(),
+				},
+				Validators: []tfsdk.AttributeValidator{
+					validators.URLScheme("http", "https", "ssh"),
+				},
+			},
+			"interval": {
+				Description: "Interval at which to reconcile from bootstrap repository.",
+				Type:        customtypes.DurationType,
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					planmodifiers.DefaultValue(customtypes.DurationValue(time.Minute)),
+				},
+			},
+			"path": {
+				Description: "Path relative to the repository root, when specified the cluster sync will be scoped to this path",
+				Type:        types.StringType,
+				Optional:    true,
+			},
+			"branch": {
+				Description: "Branch in repository to reconcile from.",
+				Type:        types.StringType,
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					resource.RequiresReplace(),
+					planmodifiers.DefaultStringValue("main"),
+				},
+			},
+			"recurse_submodules": {
+				Description: "",
+				Type:        types.BoolType,
+				Optional:    true,
+			},
+			"secret_name": {
+				Description: "",
+				Type:        types.StringType,
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					resource.RequiresReplace(),
+					planmodifiers.DefaultStringValue(defaultOpts.Namespace),
+				},
+				Validators: []tfsdk.AttributeValidator{
+					stringvalidator.RegexMatches(regexp.MustCompile(rfc1123DomainRegex), rfc1123DomainError),
+					stringvalidator.LengthAtMost(253),
+				},
+			},
+			"kustomization_override": {
```

Comment:
Is this  a reasonable way to implement customizing of Flux?

Reply:
@stefanprodan I just wanted to check to see if you have any opinions on this design choice. Basically allowing end users to write their own kustomization file which will override the one generated during bootstrap. This should solve all "custom" requirements as the changes will apply both on first apply and when being reconciled later by Flux.